package/network/config/gre/files/gre.sh

```diff
@@ -291,8 +291,6 @@ proto_grev6tap_init_config() {
 }
 
 [ -n "$INCLUDE_ONLY" ] || {
-	[ -f /lib/modules/$(uname -r)/gre.ko ] && add_protocol gre
-	[ -f /lib/modules/$(uname -r)/gre.ko ] && add_protocol gretap
-	[ -f /lib/modules/$(uname -r)/ip6_gre.ko ] && add_protocol grev6
-	[ -f /lib/modules/$(uname -r)/ip6_gre.ko ] && add_protocol grev6tap
+	[ -f /lib/modules/$(uname -r)/gre.ko ] || [ -d /sys/module/ip_gre ] && { add_protocol gre; add_protocol gretap; }
+	[ -f /lib/modules/$(uname -r)/ip6_gre.ko ] || [ -d /sys/module/ip6_gre ] && { add_protocol grev6; add_protocol grev6tap; }
```

Comment:
I'd argue checking for the existence of the module files is completely wrong (think monolithic builds) and shouldn't even be done…